src/fancy-transformers/fragment.ts

```diff
@@ -0,0 +1,177 @@
+import logging = require('../logging/logging');
+
+var log :logging.Log = new logging.Log('fancy-transformers');
+
+var takeByte_ = (buffer:ArrayBuffer) : number => {
+  var bytes=new Uint8Array(buffer);
+  return bytes[0];
+}
+
+var dropByte_ = (buffer:ArrayBuffer) : ArrayBuffer => {
+  var bytes=new Uint8Array(buffer)
+  var result = new Uint8Array(buffer.byteLength-1);
+  var fromIndex : number = 1;
+  var toIndex : number = 0;
+  while(toIndex < result.length) {
+    result[toIndex] = bytes[fromIndex];
+    toIndex=toIndex+1;
+    fromIndex=fromIndex+1;
+  }
+
+  return result.buffer;
+}
+
+var encodeByte_ = (num:number) : ArrayBuffer => {
+  var bytes = new Uint8Array(1);
+  bytes[0]=num;
+  return bytes.buffer;
+}
+
+var split_ = (buffer:ArrayBuffer, firstLen:number) : Array<ArrayBuffer> => {
+  var bytes=new Uint8Array(buffer)
+  var lastLen : number = buffer.byteLength-firstLen;
+  var first = new Uint8Array(firstLen);
+  var last = new Uint8Array(lastLen);
+  var fromIndex : number = 0;
+  var toIndex : number = 0;
+  while(toIndex < first.length) {
+    first[toIndex] = bytes[fromIndex];
+    toIndex=toIndex+1;
+    fromIndex=fromIndex+1;
+  }
+
+  toIndex=0;
+  while(toIndex < last.length) {
+    last[toIndex] = bytes[fromIndex];
+    toIndex=toIndex+1;
+    fromIndex=fromIndex+1;
+  }
+
+  return [first.buffer, last.buffer];
+}
+
+/* Takes a two byte (network byte order) representation of a number and returns
+ * the number.
+ */
+ // TODO(bwiley): Byte order may be backward
+ // TODO(bwiley): Fix type error
+var decodeLength_ = (buffer:ArrayBuffer) : number => {
```

Comment:
Perhaps generalize to ntohs/htons()?  `ntohs(buffer:ArrayBuffer, offset:int) :number` and `htons(value:number, buffer:ArrayBuffer, offset:int) :void`?